addons/mail/static/src/models/message/message.js

```diff
@@ -51,12 +51,18 @@ function factory(dependencies) {
                     // partner id 0 is a hack of message_format to refer to an
                     // author non-related to a partner. display_name equals
                     // email_from, so this is omitted due to being redundant.
-                    data2.author = [
-                        ['insert', {
+                    let author = {};
+                    if (Array.isArray(data.author_id)) {
+                        author = {
                             display_name: data.author_id[1],
                             id: data.author_id[0],
-                        }],
-                    ];
+                        };
+                    } else {
+                        author = {
+                            id: data.author_id,
+                        };
```

Comment:
What flux/rpc result are sending `author_id` as just id instead of (id, display_name)? It might be worth it to keep those in mind and consider changing them in master.

Reply:
I don't remember. But since those bugs have been found via tests, it could also be just a wrong test.
I trusted the test data on this one.

Reply:
Indeed, need to double check the consistency of those tests then. It might be some formatters in the mock server don't properly return the name_get result and just return the raw field value instead (which is just id).